integration-tests/chopsticks/src/tests/xcm.test.ts

```diff
@@ -0,0 +1,72 @@
+import { test, beforeAll, afterAll } from 'vitest'
+import { connectParachains, connectVertical, xcmLogger } from '@acala-network/chopsticks'
+import { sendTransaction, withExpect } from '@acala-network/chopsticks-testing'
+
+import * as SpiritnetNetwork from '../network/spiritnet.js'
+import * as PolkadotNetwork from '../network/polkadot.js'
+import * as HydraDxNetwork from '../network/hydroDx.js'
+import type { Config } from '../network/types.js'
+import { keysBob, keysCharlie } from '../helper.js'
+
+let spiritnetContext: Config
+let hydradxContext: Config
+let polkadotContext: Config
+
+beforeAll(async () => {
+	xcmLogger.level = 'info'
+	spiritnetContext = await SpiritnetNetwork.getContext()
+	hydradxContext = await HydraDxNetwork.getContext()
+	polkadotContext = await PolkadotNetwork.getContext()
+
+	await polkadotContext.dev.setStorage(PolkadotNetwork.defaultStorage(keysCharlie.address))
+	await spiritnetContext.dev.setStorage(SpiritnetNetwork.defaultStorage(keysBob.address))
+	await hydradxContext.dev.setStorage(HydraDxNetwork.defaultStorage(keysBob.address))
+
+	// Setup network
+	await connectVertical(polkadotContext.chain, spiritnetContext.chain)
+	await connectVertical(polkadotContext.chain, hydradxContext.chain)
+	await connectParachains([spiritnetContext.chain, hydradxContext.chain])
+
+	// fixes api runtime disconnect warning
+	await new Promise((r) => setTimeout(r, 500))
+	// Perform runtime upgrade
+	await Promise.all([polkadotContext.dev.newBlock(), spiritnetContext.dev.newBlock(), hydradxContext.dev.newBlock()])
```

Comment:
Why do we need to produce a block also for HydraDX? Is that required by the `connectParachains` and `connectVertical`?

Reply:
Yes, a block needs to be produced for the updated XCM connections. I changed the comment in: [58b6caa](https://github.com/KILTprotocol/kilt-node/pull/614/commits/58b6caa6a3cc3253a8573b878263b23f3c8c7c93)